src/main/java/com/example/library/model/User.java

```diff
@@ -77,7 +79,9 @@ public boolean isEnabled() {
 
     @Override
     public Collection<? extends GrantedAuthority> getAuthorities() {
-        return roles;
+        return roles.stream()
+                .map(role -> new SimpleGrantedAuthority("ROLE_" + role.getName().name()))
+                .collect(Collectors.toSet());

```

Comment:
you can user ROLE_USER in enum Role (for admin the same)

```suggestion
        return roles;
```